packages/core/src/internal/fetchOptions.ts

```diff
@@ -129,9 +129,24 @@ export const makeFetchOptions = (
     (typeof operation.context.fetchOptions === 'function'
       ? operation.context.fetchOptions()
       : operation.context.fetchOptions) || {};
-  if (extraOptions.headers)
-    for (const key in extraOptions.headers)
-      headers[key.toLowerCase()] = extraOptions.headers[key];
+  if (extraOptions.headers) {
+    if (extraOptions.headers.forEach) {
+      (extraOptions.headers as Headers | [[string, string]]).forEach(
+        (value, key) => {
+          if (Array.isArray(value)) {
+            headers[value[0]] = value[1];
```

Comment:
This doesn't seem right and should be comma-separated 🤔

Reply:
Oh, sorry, I meant, `[string, string[]]`, but I'm not sure whether that'll actually be issued as repeated `[string, string]` entries. Sorry!

Reply:
According to the `HeadersInit` type, which we support here it's `[[string, string]]` which I guess makes it so that we have to merge into existing headers